woocommerce/payment-gateway/payment-tokens/class-sv-wc-payment-gateway-payment-token.php

```diff
@@ -538,6 +538,81 @@ private function read( \WC_Payment_Token $core_token ) {
 	}
 
 
+	/**
+	 * Stores the token data in the database.
+	 *
+	 * Stores the token data as a Woocommerce payment token, in the `wp_woocommerce_payment_tokens` table,
+	 * with meta data in the `wp_woocommerce_payment_tokenmeta` table.
+	 *
+	 * @since 5.6.0-dev.1
+	 */
+	public function save() {
+
+		if ( $this->token instanceof \WC_Payment_Token ) {
+			$token = $this->token;
+		} else {
+
+			// see if there is already a token with this ID saved for this customer and gateway
+			// TODO: use `get_user_id` and `get_gateway_id`, once merged {DM 2019-12-12}
+			$saved_tokens = \WC_Payment_Tokens::get_customer_tokens( $this->data['user_id'], $this->data['gateway_id'] );
+
+			foreach ( $saved_tokens as $saved_token ) {
+
+				if ( $saved_token->get_token() === $this->get_id() ) {
+					$token = $saved_token;
+					break;
+				}
+			}
+
+			if ( empty( $token ) ) {
+
+				// instantiate a new token
+				if ( $this->is_credit_card() ) {
+					$token = new \WC_Payment_Token_CC();
+				} elseif ( $this->is_echeck() ) {
+					$token = new \WC_Payment_Token_ECheck();
+				}
+
+				// mark the token as migrated
+				$this->data['migrated'] = true;
+
+				// update legacy token in user meta data
+				$gateways   = WC()->payment_gateways->payment_gateways();
+				// TODO: use `get_gateway_id`, once merged {DM 2019-12-12}
+				$gateway_id = $this->data['gateway_id'];
+
+				if ( ! empty( $gateway_id ) && ! empty( $gateway = $gateways[ $gateway_id ] ) ) {
+					// TODO: use `get_user_id`, once merged {DM 2019-12-12}
+					$gateway->get_payment_tokens_handler()->update_legacy_token( $this->data['user_id'], $this );
+				}
+			}
+
+			if ( ! empty( $token ) ) {
+
+				$token->set_token( $this->get_id() );
+
+				foreach ( $this->data as $key => $value ) {
+
+					if ( 'exp_year' === $key && 2 === strlen( $value ) ) {
+						$value = '20' . $value;
+					}
+
+					$core_key = array_search( $key, $this->props );
+
+					if ( false !== $core_key ) {
+						$token->set_props( [ $core_key => $value ] );
+					} else {
+						// metadata
```

Comment:
Comment seems unnecessary when you have a method that says `add_meta_data()` :wink: 

More importantly, should this be `add_meta_data()` or `update_meta_data()`? Because the `save()` method could be used to update a token as well - we don't want skips for existing data (since `unique` is set to `true`).